### PR TITLE
Introduced new number_of_retries option to retry after failure

### PR DESCRIPTION
**Pull request by KrauseFx**
- To workaround flaky UI Tests
- New fastlane UI

Fixes https://github.com/fastlane/snapshot/issues/372

### CODE REVIEW

File: lib/snapshot/runner.rb [hunk 1]
```diff
@@ -55,6 +51,24 @@ def work
       FileUtils.rm_rf(TestCommandGenerator.derived_data_path)
     end
 
+    # This is its own method so that it can re-try if the tests fail randomly
+    # @return true/false depending on if the tests succeded
+    def run_for_device_and_language(language, device, launch_arguments, retries = 0)
+      return launch(language, device, launch_arguments)
+    rescue => ex
+      UI.error ex.to_s # we should to show right here as well
```
Comment (i2amsam) [inline lib/snapshot/runner.rb]: I don't follow this comment.  Stale?

File: lib/snapshot/runner.rb [hunk 2]
```diff
@@ -55,6 +51,24 @@ def work
...
+
+      if retries < Snapshot.config[:number_of_retries]
+        UI.important "Tests failed, re-trying #{retries + 1} out of #{Snapshot.config[:number_of_retries] + 1} times"
+        run_for_device_and_language(language, device, launch_arguments, retries + 1)
+      else
+        UI.error "Backtrace:\n\t#{ex.backtrace.join("\n\t")}" if $verbose
+        self.collected_errors << ex
+        raise ex if Snapshot.config[:stop_after_first_error]
```
Comment (i2amsam) [inline lib/snapshot/runner.rb]: These two commandline options now seem a bit confusing.  I'm not sure how to solve it.  This option would probably be better as `stop_after_first_complete_failure` and the new one `number_of_retries_for_failed_test` but changing the existing option would not be backward compatible.

Comment (KrauseFx) [inline lib/snapshot/runner.rb]: Very good point, however we can't really rename the other option. I'm more than happy to make the documentation on both options more clear :+1:

**Comment by KrauseFx** [comment]
@i2amsam except for you 2 comments, the rest looks okay to you? :+1:

**Comment by i2amsam** [comment]
Yep, looks :100: to me.  :rocket: it

**Comment by KrauseFx** [comment]
Awesome, thank you. I'll send another PR for the updated docs

**Pull request merged**